src/main/java/com/google/cloud/dns/Dns.java

```diff
@@ -380,6 +413,81 @@ public static ChangeRequestListOption sortOrder(SortingOrder order) {
     }
   }
 
+  /** Class for specifying DNS request field options. */
+  class DnsKeyOption extends Option {
+
+    DnsKeyOption(DnsRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify which fields of {@link DnsKeyInfo} should be returned by the
+     * service.
+     *
+     * <p>If this option is not provided, all change request fields are returned. {@code
+     * DnsKeyField.fields} can be used to specify only the fields of interest. The ID of the change
+     * request is always returned, even if not specified. {@link DnsKeyField} provides a list of
+     * fields that can be used.
+     */
+    public static DnsKeyOption fields(DnsKeyField... fields) {
```

Comment:
public method inside nonpublic class

Reply:
@elharo the current implementations is based on existing code to maintain consistency but if you want to change methods signature as like you mentioned then it will be different from existing code and it won't be maintain consistency.

Reply:
One more time: It is explicit google Java policy not to maintain consistency with existing code in new code when the existing code violates coding standards. There might be a reason to make this method public that I'm not seeing, but that some other existing methods are incorrectly public is not one of them.